tests/test_request_id.py

```diff
@@ -1,77 +1,453 @@
+        },
+    ),
+    (
+        # extra_config
+        {
+            # not setting DM_REQUEST_ID_HEADER should cause it to fall back to the default, DM-Request-ID
+            "DM_DOWNSTREAM_REQUEST_ID_HEADER": "DOWNSTREAM-REQUEST-ID",
+        },
+        # extra_req_headers
+        (
+            ("x-b3-traceid", "H. M. S. Belleisle",),  # should be ignored as default header name has been overwritten
+        ),
+        # expected_trace_id
+        _GENERATED_TRACE_VALUE,
+        # expect_uuid_call
+        True,
+        # expected_onwards_req_headers
+        {
+            "DM-Request-ID": _GENERATED_TRACE_VALUE,
+            "DOWNSTREAM-REQUEST-ID": _GENERATED_TRACE_VALUE,
+        },
+        # expected_resp_headers
+        {
+            "DM-Request-ID": _GENERATED_TRACE_VALUE,
+            "DOWNSTREAM-REQUEST-ID": _GENERATED_TRACE_VALUE,
+        },
+    ),
+    (
+        # extra_config
+        {
+            "DM_TRACE_ID_HEADERS": ("x-tommy-caffrey", "y-jacky-caffrey",),
+            "DM_REQUEST_ID_HEADER": "DM-REQUEST-ID",
+            "DM_DOWNSTREAM_REQUEST_ID_HEADER": "DOWNSTREAM-REQUEST-ID",
+        },
+        # extra_req_headers
+        (
+            # these should both be ignored because of the presence of the DM_TRACE_ID_HEADERS setting
+            ("DM-REQUEST-ID", "from-header",),
+            ("DOWNSTREAM-REQUEST-ID", "from-downstream",),
+        ),
+        # expected_trace_id
+        _GENERATED_TRACE_VALUE,
+        # expect_uuid_call
+        True,
+        # expected_onwards_req_headers
+        {
+            "x-tommy-caffrey": _GENERATED_TRACE_VALUE,
+            "y-jacky-caffrey": _GENERATED_TRACE_VALUE,
+        },
+        # expected_resp_headers
+        {
+            "x-tommy-caffrey": _GENERATED_TRACE_VALUE,
+            "y-jacky-caffrey": _GENERATED_TRACE_VALUE,
+        },
+    ),
+    (
+        # extra_config
+        {
+            "DM_TRACE_ID_HEADERS": ("x-tommy-caffrey", "y-jacky-caffrey",),
+        },
+        # extra_req_headers
+        (
+            ("y-jacky-caffrey", "jacky-header-value",),
+            ("x-tommy-caffrey", "tommy-header-value",),
+        ),
+        # expected_trace_id
+        "tommy-header-value",
+        # expect_uuid_call
+        False,
+        # expected_onwards_req_headers
+        {
+            "x-tommy-caffrey": "tommy-header-value",
+            "y-jacky-caffrey": "tommy-header-value",
+        },
+        # expected_resp_headers
+        {
+            "x-tommy-caffrey": "tommy-header-value",
+            "y-jacky-caffrey": "tommy-header-value",
+        },
+    ),
+    (
+        # extra_config
+        {
+            "DM_REQUEST_ID_HEADER": "DM-REQUEST-ID",
+            # not setting DM_DOWNSTREAM_REQUEST_ID_HEADER should cause it to fall back to the default, X-B3-TraceId
+        },
+        # extra_req_headers
+        (
+            ("x-kidneys", "pork",),
+            ("x-b3-traceid", "Grilled Mutton",),
```

Comment:
If the uppercasing behaviour is something we are relying on, please could it be either mentioned as an inline comment or split into a separate test please just so it's clear that it's intentional behaviour that we are testing and not just an unrelated way of how our tests are written.

Reply:
- [ ] TODO